From 6e2a815fa13ba71f58d7418507af2c751090e97c Mon Sep 17 00:00:00 2001
From: Bogdan Togorean <bogdan.togorean@analog.com>
Date: Thu, 11 Mar 2021 11:06:51 +0200
Subject: [PATCH 1/5] [PATCH 1/2] Enable I2C in PMIC

Signed-off-by: TalPilo <tal.pilo@solid-run.com>
---
 drivers/mfd/pca9450.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/drivers/mfd/pca9450.c b/drivers/mfd/pca9450.c
index 1586852ce52d..7d132afa4d95 100644
--- a/drivers/mfd/pca9450.c
+++ b/drivers/mfd/pca9450.c
@@ -235,6 +235,13 @@ static int pca9450_i2c_probe(struct i2c_client *i2c,
 			__func__);
 		goto err;
 	}
+
+	ret = pca9450_reg_write(pca9450, PCA9450_CONFIG2, 11);
+	if (ret < 0) {
+                dev_err(pca9450->dev, "%s(): write PCA9450_CONFIG2 failed!\n",
+                        __func__);
+                goto err;
+        }
 	dev_info(pca9450->dev, "Device ID=0x%X\n", ret);
 
 	pca9450_irq_init(pca9450, of_pmic_plat_data);
@@ -294,6 +301,7 @@ static int __init pca9450_i2c_init(void)
 /* init early so consumer devices can complete system boot */
 subsys_initcall(pca9450_i2c_init);
 
+
 static void __exit pca9450_i2c_exit(void)
 {
 	i2c_del_driver(&pca9450_i2c_driver);
-- 
2.17.1

